rust/src/base.rs

```diff
@@ -152,14 +148,14 @@ impl Base {
         }
         .to_string();
 
-        Ok(Base {
-            id: Arc::new(base_id),
-            position: Arc::new(road_network.position_from_geoid(h3cell.to_string())),
-            membership: Arc::new(Membership::default()),
-            total_stalls: Arc::new(stall_count),
-            available_stalls: Arc::new(stall_count),
-            station_id: Arc::new(Some(station_id)),
-        })
+        Ok(Base::new(
```

Comment:
ya, i see the value here of pushing the Arc::new stuff into the custom constructor you wrote. nice.